### PR TITLE
updating refs, use external script

### PR DESCRIPTION
**Pull request by e0d**
@edx/devops @cpennington 

Simplifies the common builds by using a shared external script.  Updates refs to master now that everything has merged.

### CODE REVIEW

File: docker/build/precise-common/Dockerfile [hunk 1]
```diff
@@ -2,53 +2,8 @@ FROM ubuntu:precise
...
-RUN sudo git checkout ${CONFIGURATION_VERSION}
-RUN sudo pip install -r pre-requirements.txt
-RUN sudo pip install -r requirements.txt
-WORKDIR /tmp/configuration/playbooks/edx-east
-RUN sudo /tmp/ansible/bin/ansible-playbook edx_ansible.yml -i '127.0.0.1,' -c local -e "configuration_version=${CONFIGURATION_VERSION}"
-WORKDIR /edx/app/edx_ansible
-# cleanup
-RUN sudo rm -rf /tmp/ansible
-RUN sudo rm -rf /tmp/configuration
+ADD https://raw.githubusercontent.com/edx/configuration/master/util/install/ansible-bootstrap.sh /tmp/ansible-bootstrap.sh
```
Comment (cpennington) [inline docker/build/precise-common/Dockerfile]: Can we `ADD` this from the local directory, so that it's possible to actually do local testing/debugging?

Comment (cpennington) [inline docker/build/precise-common/Dockerfile]: Especially with the changes in the #2574, which would let us do pushes from CI (so the risk of a dirty local repo is significantly less).

Comment (e0d) [inline docker/build/precise-common/Dockerfile]: @cpennington this isn't really possible.
`ADD` doesn't all fq or relative references to files outside the the build context and also doesn't support symlinks.  It does support URLs but not file URLs, e.g., `file:///tmp/file.txt`

For debugging, though a pain, I think the best we can do is bash onto a bare server and run the file locally.

It's a pain, but I think the reuse is worth the trade-off.  Obviously providing access outside the build directory is not hard to do, but has significant security implications for folks building from provided Dockerfiles.

**Comment by feanil** [comment]
Seems like a good incremental improvement. :+1:

**Comment by e0d** [comment]
@cpennington updated to use `-f` strategy.

File: docker/build/precise-common/Dockerfile [hunk 2]
```diff
@@ -2,53 +2,8 @@ FROM ubuntu:precise
 MAINTAINER edxops 
 ENV ANSIBLE_REPO="https://github.com/edx/ansible"
 ENV CONFIGURATION_REPO="https://github.com/edx/configuration.git"
-ENV CONFIGURATION_VERSION="e0d/docker-artifacts"
+ENV CONFIGURATION_VERSION="master"
```
Comment (cpennington) [inline docker/build/precise-common/Dockerfile:5]: Do you need this anymore?

Comment (e0d) [inline docker/build/precise-common/Dockerfile:5]: For now, yes.  May pull your suggestion in from the other PR later.

Comment (cpennington) [inline docker/build/precise-common/Dockerfile:5]: Oh, is that environment variable used inside the `ansible-bootstrap.sh` script?

Comment (e0d) [inline docker/build/precise-common/Dockerfile:5]: Yes.

**Comment by e0d** [comment]
@cpennington OK to merge?

**Comment by cpennington** [comment]
:+1:

File: docker/build/precise-common/Dockerfile [hunk 3]
```diff
@@ -2,53 +2,8 @@ FROM ubuntu:precise
...
 
-# system bootstrap
-RUN apt-get update
-RUN apt-get -y install software-properties-common python-software-properties
-RUN add-apt-repository ppa:fkrull/deadsnakes-python2.7
-RUN apt-get update
-RUN apt-get -y install sudo
-RUN useradd docker && echo "docker:docker" | chpasswd
-RUN echo "docker ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
-RUN mkdir -p /home/docker && chown -R docker:docker /home/docker
```
Comment (cpennington) [inline docker/build/precise-common/Dockerfile:15]: I just noticed that you removed the stuff that set up the docker user. Was that deliberate? Should our dockerfiles switch to just using the root user?

**Pull request merged**